src/edu/stanford/nlp/ie/NERFeatureFactory.java

```diff
@@ -1579,85 +1670,84 @@ else if (flags.useTitle2) {
         makeGenericKeyCache(c);
       }
       //now look through the cached keys
-      for (Class key : genericAnnotationKeys) {
+      for (Class<?> key : genericAnnotationKeys) {
         //log.info("Adding feature: " + CoreLabel.genericValues.get(key) + " with value " + c.get(key));
-        if (c.get(key) != null && c.get(key) instanceof Collection) {
-          for (Object ob: (Collection)c.get(key)) {
-            featuresC.add(ob + "-" + CoreLabel.genericValues.get(key));
+        Object col = c.get((Class<CoreAnnotation<Object>>) key);
+        if (col instanceof Collection) {
+          for (Object ob: (Collection<?>)col) {
+            out.addConcat(ob.toString(), "-", CoreLabel.genericValues.get(key));
           }
         } else {
-          featuresC.add(c.get(key) + "-" + CoreLabel.genericValues.get(key));
+          out.addConcat(col.toString(), "-", CoreLabel.genericValues.get(key));
```

Comment:
Is there a bug here, originally from the old version?  If you have a null value, the old version would add the feature null-key and the new version will NPE

Reply:
I'll add `if (col != null)` but I do not know whether it was intentional to have null-key features there.

Reply:
I don't think this code is commonly used, and in fact I also do not think it will work reliably, because of the `makeGenericKeyCache`, which will only populated once, and anything added afterwards will likely not work. Maybe you can discuss whether this can be removed. In CoreLabel some support for this seems to be already commented out.